Content.Server/Cluwne/CluwneSystem.cs

```diff
@@ -0,0 +1,140 @@
+using Content.Server.Administration.Commands;
+using Content.Server.Popups;
+using Content.Shared.Popups;
+using Content.Server.Interaction.Components;
+using Content.Shared.Mobs;
+using Content.Server.Chat.Systems;
+using Robust.Shared.Random;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Stunnable;
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.Damage;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timing;
+using Content.Server.Emoting.Systems;
+using Content.Server.Speech.EntitySystems;
+using Content.Server.Database;
+using Content.Shared.Cluwne;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Content.Server.Cluwne;
+
+public sealed class CluwneSystem : EntitySystem
+{
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly SharedStunSystem _stunSystem = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+
+
+        SubscribeLocalEvent<CluwneComponent, ComponentStartup>(OnComponentStartup);
+        SubscribeLocalEvent<CluwneComponent, MobStateChangedEvent>(OnMobState);
+        SubscribeLocalEvent<CluwneComponent, EmoteEvent>(OnEmote, before:
+            new[] { typeof(VocalSystem), typeof(BodyEmotesSystem) });
+
+    }
+
+    public override void
+
+     Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var cluwnecomp in EntityQuery<CluwneComponent>())
+        {
+            cluwnecomp.LastGiggleCooldown -= frameTime;
+
+            RandomEmote(cluwnecomp.Owner, cluwnecomp);
+        }
+
```

Comment:
As I said before random emotes should be shared with zombies.

Reply:
Yea, i have been trying to get this to work but a couple of people helping me also were stumped or gave up on this. Will keep at it though.

Reply:
Cluwne now uses the autoemote system so this should be resolved.